### ISSUE TITLE
Error opening project in Android Studio after adding Firebase Analytics 6.14

### ISSUE THREAD

**Issue by stopiccot** [firebase/quickstart-unity#655]
### Please fill in the following fields:
Unity editor version: 2018.4.16f1
Firebase Unity SDK version: 6.14
Source you installed the SDK: .unitypackage
Firebase plugins in use: Analytics
Additional SDKs you are using: none
Platform you are using the Unity editor on: Mac
Platform you are targeting: Android
Scripting Runtime: Mono
Android Studio: 3.5.2

### Please describe the issue here:
I use Export Project option in Unity to generate Android Studio project for my game.
After adding Firebase Analytics to my project I get an error while opening:
```
ERROR: Manifest merger failed : uses-sdk:minSdkVersion 1 cannot be smaller than version 14 declared in library [__tested_artifact__::Firebase] /Users/stopiccot/code/builds/firebase_test/firebase_test/Firebase/build/intermediates/library_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 1
	Suggestion: use a compatible library with a minSdk of at most 1,
		or increase this project's minSdk version to at least 14,
		or use tools:overrideLibrary="com.google.firebase.unity" to force usage (may lead to runtime failures)
```
Everything worked fine before adding Firebase Analytics

Test project can be found at https://github.com/stopiccot/firebase_test

**Comment by DellaBitta** [comment]
I attempted to reproduce this using the Firebase Analytics quickstart with Android Studio 3.6.3 and Unity Version 2019.2.12f1, importing Firebase Analytics 6.14.0 from the .unitypackage files (not through the package manager), but everything seemed to work out for me.

One thing I noticed is that the **Assets/manifest.json** file in your github repo doesn't have an entry for Firebase Analytics:  

```
"dependencies": {
    "com.google.firebase.analytics": "6.14.0",
    ...
}
```
Perhaps attempt a re-import of the library?  Thanks!

**Comment by stopiccot** [comment]
Correct me if I'm wrong but **Packages/manifest.json** is used for UPM (Unity Package Manager) dependencies. I'm using unitypackage version instead.

**Comment by stopiccot** [comment]
The only difference after re-importing FirebaseAnalytics.unitypackage is .so files from Assets/Firebase/Plugins/x86_64 that I've removed so I can fit to 100mb per file github limitation

**Comment by chkuang-g** [comment]
@stopiccot 

You are correct.  `Packages/manifest.json` is only used for UPM and does not seems to be related to you case.

I downloaded your project and try to build it with Unity 2018.4.22f1 and I did not have any trouble to build the apk.  So I guess this is some issue with Android tools or build tools.

I would recommend you go to the SDK Tool tab in SDK Manager in Android Studio
- Check on `Hide Obsolete Packages` at the bottom of the window and make sure `Android SDK Tools (Obsolete)` is installed
- Update Android SDK Build-Tool to non-RC version.  You can check `Show Package Details` at the bottom of the window to show non-rc versions of build tools.  Ex. by 06/01/2020 the latest non-rc version is 29.0.3.

Hope this helps.
Shawn

**Comment by stopiccot** [comment]
I've updated build tools to 29.0.3
Now I've started to get error:
```
ERROR: Failed to resolve: firebase-analytics-unity
Affected Modules: firebase_test
```
I really have no idea why it's happening. Path local maven repo looks to be correct:
```
maven {
    url "file:////Users/stopiccot/firebase_test/Assets/Firebase/m2repository" // Assets/Firebase/Editor/AnalyticsDependencies.xml:18, Assets/Firebase/Editor/AppDependencies.xml:20
}
```

**Comment by stopiccot** [comment]
@chkuang-g Please note that building apk directly from Unity Editor works ok. The problem occurs when I try to export android studio project and later build it using android studio

**Comment by chkuang-g** [comment]
Ah, interesting.  I think you _might_ be experiencing https://github.com/googlesamples/unity-jar-resolver/issues/366

Could you try the following approaches:

- Go to `Assets > External Dependency Manager > Android Resolver > Settings` menu item.  Make sure `Patch mainTemplate.gradle` is checked and see what is current path for `Local Maven Repo Directory`, ex. `Assets/GeneratedLocalRepo` by default.
- See if there is any `.aar` files generated under the path for `Local Maven Repo Directory`.
  - If there is any, that might be related to https://github.com/googlesamples/unity-jar-resolver/issues/366.  In that case, modify your `mainTemplate.gradle` from:
  ```
  maven {
    url "file:////Users/stopiccot/firebase_test/Assets/Firebase/m2repository" // Assets/Firebase/Editor/AnalyticsDependencies.xml:18, Assets/Firebase/Editor/AppDependencies.xml:20
  }
  ```
  To
  ```
  maven {
      url (unityProjectPath + "/Assets/GeneratedLocalRepo/Firebase/m2repository")
  }
   
  ```

**Comment by stopiccot** [comment]
Oh, local repo again... I pray to all known to me gods that one day Firebase developers will just upload all aars to maven instead of maintaining this troublesome buggy local repo thing :)

**Comment by stopiccot** [comment]
Manually modifying gradle file after each export is not an option for me cause it will break CI process.

**Comment by stopiccot** [comment]
But yes, manually editing file looks to be fixing issue

**Comment by chkuang-g** [comment]
This should be fixed in EDM4U 1.2.156.

https://github.com/googlesamples/unity-jar-resolver/raw/master/external-dependency-manager-1.2.156.unitypackage

Let us know if this resolved your issue.

**Comment by stopiccot** [comment]
Seems to be resolved in Firebase 6.15.1
